### DISCUSSION TITLE
Use Towncrier?

### DISCUSSION THREAD

**Discussion by alessio-locatelli** [requests-cache/aiohttp-client-cache#204] [Ideas]
- https://github.com/twisted/towncrier

I have no strong opinion on it (I do not want to say that a project can not be maintained without it), but I am using it for a year and I like that:
- not required to edit the changelog manually
- changes are nicely grouped (e.g. `Misc`, `Features`, `Bugfixes`)

Since Towncrier usage example is documented in  https://docs.aiohttp.org/en/stable/contributing.html#changelog-update, it may be possible just to put a link to the `CONTRIBUTING.md` file.

**Comment by JWCook** [comment]
I am familiar with towncrier, but haven't used it on any projects yet. My initial impressions are:
* For a smaller project like this, adding news fragments seems like about as much effort as adding lines to a changelog.
* For a larger project with several regular contributors, the most obvious benefit seems like avoiding merge conflicts on a single changelog file being edited in (almost) every PR.
* I also like that it enforces a consistent format grouped by change type.

Overall I'm not sure if this would make release notes much better or simpler for this project, but I could likely change my mind after trying it out. I'd be willing to use it for one or two minor releases, and decide whether or not to keep it after that. What do you think?

**Reply by alessio-locatelli** [reply]
Not a replacement for `towncrier`, but at least we can automatically format and clean up Markdown files.  I use https://prettier.io/ for a long time in all projects. I can add it as a `pre-commit` hook.

**Reply by JWCook** [reply]
Sure! Sounds good to me.
